runtime/src/accounts_db.rs

```diff
@@ -5107,7 +5109,7 @@ impl AccountsDb {
                                 self.accounts_index.get(pubkey, Some(ancestors), Some(slot))
                             {
                                 let (slot, account_info) = &lock.slot_list()[index];
-                                if account_info.lamports != 0 {
+                                if true { // account_info.lamports != 0 {
```

Comment:
saving this short circuit/avoiding the account doesn't seem that big of a deal, since we don't use this accounts hashing path anymore anyways

Reply:
yes, this short circuit was hacked in because I knew this path wasn't critical atm. I just cleaned it up so it is correct and not hacky.